### ISSUE TITLE
G83 - Is there a need to meet 4.1.2 Name, Role, Value?

### ISSUE THREAD

**Issue by anevins12** [w3c/wcag#3365]
Hello!

Does sufficient [technique G83](https://www.w3.org/WAI/WCAG21/Techniques/general/G83) require the errors presented in the alert dialog to be programmatically associated with their respective input fields for 4.1.2 Name, Role, Value?

I'm not too interested in how this will be technically implemented. I'm more interested in whether providing an alert dialog to present error messages fails 4.1.2 if those messages are not programmatically associated to their form fields.

**Comment by patrickhlauke** [comment]
if by dialog you mean an actual modal dialog ... there's no sense in associating it with the input field in the underlying page, as that won't be exposed/visible when you're inside the little self-contained world of the dialog. or am i misunderstanding?

(and the usual "a technique doesn't *require* anything, it's only an informative suggestion/idea)

**Comment by anevins12** [comment]
> if by dialog you mean an actual modal dialog ... there's no sense in associating it with the input field in the underlying page, as that won't be exposed/visible when you're inside the little self-contained world of the dialog. or am i misunderstanding?

I mean a modal dialog. That's right, but I was thinking for the benefit of people closing the modal dialog and returning to the form in error. When the modal dialog presents error messages, I think it would be useful to also programmatically associate those error messages so that they are available for people navigating the form (after closing the modal dialog.)

**Comment by scottaohara** [comment]
Sounds like the dialog is not the right way to surface these errors then, if you also want the errors available after one dismisses the dialog

**Comment by anevins12** [comment]
I agree, although a normative failure is better received than a best practice recommendation. If a customer implements error messages in a modal dialog, can I say that this insufficient to meet 4.1.2?

**Comment by anevins12** [comment]
At this point, I don't know whether providing error messages in a modal dialog will pass 4.1.2 and that's what I'm trying to find out.
